source/common/common/assert.cc

```diff
@@ -28,15 +28,63 @@ class ActionRegistrationImpl : public ActionRegistration {
   static std::function<void()> debug_assertion_failure_record_action_;
 };
 
+class EnvoyBugRegistrationImpl : public ActionRegistration {
+public:
+  EnvoyBugRegistrationImpl(std::function<void()> action) {
+    ASSERT(envoy_bug_failure_record_action_ == nullptr);
+    envoy_bug_failure_record_action_ = action;
```

Comment:
Would it be worth adding:
count_ = 0;

The benefit is that we get an effective reset of this counter when the server is created in e2e integration tests, so we get some better determinism for ENVOY_BUG logging in opt mode.

Reply:
Done - haha, I was hitting this before in my tests and being like ... why!! :)